Backpack/BarChart/Classes/BarChartBar.swift

```diff
@@ -0,0 +1,202 @@
+            backgroundView.centerXAnchor.constraint(equalTo: centerXAnchor),
+            miniFlareView.centerXAnchor.constraint(equalTo: backgroundView.centerXAnchor),
+            titleLabel.centerXAnchor.constraint(equalTo: backgroundView.centerXAnchor),
+            subtitleLabel.centerXAnchor.constraint(equalTo: backgroundView.centerXAnchor),
+            titleLabel.topAnchor.constraint(equalTo: backgroundView.bottomAnchor, constant: BPKSpacingMd),
+            subtitleLabel.topAnchor.constraint(equalTo: titleLabel.bottomAnchor, constant: BPKSpacingMd),
+            bottomAnchor.constraint(equalTo: subtitleLabel.bottomAnchor, constant: BPKSpacingMd)
+        ])
+        updateBarHeight()
+        updateminiFlareViewAppearance()
+    }
+
+    lazy fileprivate var backgroundView: UIView = {
+        let view = UIView()
+        view.backgroundColor = BPKColor.dynamicColor(withLightVariant: BPKColor.skyGrayTint06,
+                                                     darkVariant: BPKColor.blackTint03)
+        view.translatesAutoresizingMaskIntoConstraints = false
+        view.layer.cornerRadius = BPKSpacingMd
+        return view
+    }()
+
+    lazy fileprivate var barView: UIView = {
+        let view = UIView()
+        view.backgroundColor = BPKBarChartBar.noValueColor
+        view.translatesAutoresizingMaskIntoConstraints = false
+        view.layer.cornerRadius = BPKSpacingMd
+        return view
+    }()
+
+    lazy fileprivate var miniFlareView: BPKMiniFlareView = {
+        let view = BPKMiniFlareView()
+        view.translatesAutoresizingMaskIntoConstraints = false
+        return view
+    }()
+
+    lazy fileprivate var titleLabel: BPKLabel = {
+        let view = BPKLabel(fontStyle: .textBaseEmphasized)
+        view.translatesAutoresizingMaskIntoConstraints = false
+        return view
+    }()
+
+    lazy fileprivate var subtitleLabel: BPKLabel = {
+        let view = BPKLabel(fontStyle: .textSm)
+        view.translatesAutoresizingMaskIntoConstraints = false
+        return view
+    }()
+
+    static fileprivate var noValueColor: UIColor = BPKColor.skyGrayTint03
+
+    static fileprivate var valueColor: UIColor = BPKColor.primaryColor
+
+    static fileprivate var selectedColor: UIColor = BPKColor.dynamicColor(withLightVariant: BPKColor.monteverde,
+                                                                          darkVariant: BPKColor.glencoe)
+
+    fileprivate func updateBarColor() {
+        if isSelected {
+            barView.backgroundColor = BPKBarChartBar.selectedColor
+            titleLabel.textColor = BPKBarChartBar.selectedColor
+            subtitleLabel.textColor = BPKBarChartBar.selectedColor
+            return
+        }
+
+        titleLabel.textColor = BPKColor.textPrimaryColor
+        subtitleLabel.textColor = BPKColor.textPrimaryColor
+
+        if fillValue == nil {
+            barView.backgroundColor = BPKBarChartBar.noValueColor
```

Comment:
This can be written as `barView.backgroundColor = fillValue.map { BPKBarChartBar.valueColor } ?? BPKBarChartBar.noValueColor`

which is slightly nicer